test/e2e/cli/pkg_source_test.go

```diff
@@ -0,0 +1,332 @@
+					cmd.Args = append(cmd.Args, "https://example2.com")
+					err := cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Read the config file
+					osFile, err := os.Open(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+					defer osFile.Close()
+
+					osFileInfo, err := osFile.Stat()
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file is not empty
+					Expect(osFileInfo.Size()).To(BeNumerically(">", 0))
+
+					// Check if the config file has 600 permissions
+					Expect(osFileInfo.Mode().Perm()).To(Equal(os.FileMode(0o600)))
+
+					// Read file content
+					readBytes, err := os.ReadFile(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+
+					// Marshal the yaml file into a map
+					var cfgMap map[string]interface{}
+					err = yaml.Unmarshal([]byte(readBytes), &cfgMap)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file contains the default manifests
+					cfgMapUnikernel, ok := cfgMap["unikraft"].(map[string]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Cast the manifests list to an array
+					cfgMapUnikernelManifests, ok := cfgMapUnikernel["manifests"].([]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Check if the default manifests are present and in the first position
+					Expect(cfgMapUnikernelManifests).To(HaveLen(3))
+					Expect(cfgMapUnikernelManifests[0]).To(Equal("https://manifests.kraftkit.sh/index.yaml"))
+
+					// Check if the config file contains the new link
+					Expect(cfgMapUnikernelManifests[1]).To(Equal("https://example1.com"))
+					Expect(cfgMapUnikernelManifests[2]).To(Equal("https://example2.com"))
+
+					// Check if stdout is empty
+					Expect(stdout.String()).To(BeEmpty())
+				})
+			})
+
+			When("sourcing an existing link in the config file", func() {
+				It("should warn the user and leave the file intact", func() {
+					oldArgs := make([]string, len(cmd.Args))
+					copy(oldArgs, cmd.Args)
+
+					// Generate a clean config file
+					cmd.Args = append(cmd.Args, "https://manifests.kraftkit.sh/index.yaml")
+					err := cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Calculate config file hash
+					bytes, err := os.ReadFile(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+
+					// Calculate sha1 hash of bytes
+					sha1Hash := sha1.Sum(bytes)
+
+					// Recreate the command to reuse
+					stdout = fcmd.NewIOStream()
+					stderr = fcmd.NewIOStream()
+
+					cmd = fcmd.NewKraft(stdout, stderr, cfg.Path())
+					cmd.Args = oldArgs
+
+					cmd.Args = append(cmd.Args, "https://manifests.kraftkit.sh/index.yaml")
+					err = cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Check warning message exists in stdout
+					Expect(stdout.String()).To(MatchRegexp(`^{"level":"warning","msg":"manifest already saved: https://manifests\.kraftkit\.sh/index\.yaml"}\n$`))
```

Comment:
nit: We could as well use `Equal()` here if we are gating the string between `^$`.